pkg/controllers/ippool/ippool_controller.go

```diff
@@ -134,10 +137,46 @@ func (r *IPPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 			log.V(1).Info("added finalizer on ippool CR", "ippool", req.NamespacedName)
 		}
 
+		if obj.Spec.Type == "" {
+			vpcNetworkConfig := common.ServiceMediator.GetVPCNetworkConfigByNamespace(obj.Namespace)
+			if vpcNetworkConfig == nil {
+				err := fmt.Errorf("operate failed: cannot get configuration for IPPool CR")
+				log.Error(nil, "failed to find VPCNetworkConfig for IPPool CR", "ippool", req.NamespacedName, "namespace %s", obj.Namespace)
+				updateFail(r, &ctx, obj, &err)
+				return resultRequeue, err
+			}
+			obj.Spec.Type = vpcNetworkConfig.DefaultSubnetAccessMode
+		}
+
 		subnetCidrUpdated, ipPoolSubnetsUpdated, err := r.Service.CreateOrUpdateIPPool(obj)
+		// check if ipblock is exhausted
+		apiErr, _ := util.DumpAPIError(err)
+		if apiErr != nil {
+			for _, apiErrItem := range apiErr.RelatedErrors {
+				if *apiErrItem.ErrorCode == 520012 {
+					if strings.Contains(*apiErrItem.ErrorMessage, "does not have spare capacity to satisfy") {
```

Comment:
Let's do not compare error message, NSX may change error message, error code is good enough

Reply:
I'm not sure 520012 is only for exhausted, so add an error message check to insure more safety.

Reply:
Yes. 520012 is only for exhaustion case

Reply:
OK